### ISSUE TITLE
conv1d_transpose not implemented

### ISSUE THREAD

**Issue by yaumeg** [tensorflow/tensorflow#8729]
I would like to perform a conv1d_transpose, but I can’t see any implementation in tensorflow.

I guess that inserting zeros between each values and then applying a regular conv1d would do the job ?

(principle explained hereafter with conv2d_tranpose : https://github.com/tensorflow/tensorflow/issues/4306)

But I can't find a way to implement this kind of operation with tensorflow :

import numpy as np
arr = np.arange(1, 10)                 # array([1, 2, 3, 4, 5, 6, 7, 8, 9])
np.insert(arr, slice(1, None), 0)  # array([1, 0, 2, 0, 3, 0, 4, 0, 5, 0, 6, 0, 7, 0, 8, 0, 9])

Any ideas ?

Thank you

**Comment by aselle** [comment]
Likely we will not get to this feature request quickly. I'd suggest asking on stackoverflow if someone has an elegant way to implement this. Otherwise, we'd welcome a contribution from anybody who has time to implement this. To insert zeros you might try using 
```
arr = np.arange(1, 10) # array([1, 2, 3, 4, 5, 6, 7, 8, 9])
foo = tf.zeros_like(arr)
stacked = tf.stack([arr, foo], axis=1)
zeros_inserted = tf.reshape(stacked, (stacked.get_shape().num_elements(), ))
```

**Comment by doodlesbykumbi** [comment]
@aselle I can work on this

**Comment by kleinma** [comment]
conv1d internally invokes conv2d ([here](https://github.com/tensorflow/tensorflow/blob/3836e7b94a06e1e53e77af839c6c6482565444f9/tensorflow/python/ops/nn_ops.py#L2019)). Should conv1d_transpose, similarly, invoke conv2d_tranpose?

**Comment by xXDavidHuangXx** [comment]
Hi,
I'm not very good a this, but I was wondering if you could use tf.expand_dims for a quick fix

**Comment by sdenton4** [comment]
Expanding dims for conv2d_transpose works fine for me.
One small gotcha is that your filter variables will need to be expanded on axis 0, not axis 1, since the conv2d uses the filter width, not filter height.  So:

new_x = tf.expand_dims(old_x, 1)
new_W = tf.expand_dims(old_W, 0)
deconv = tf.nn.conv2d_transpose(new_x, new_W, strides, output_shape)
deconv = tf.squeeze(deconv, axis=1)

**Comment by yongtang** [comment]
Added a PR #13105 for conv1d_transpose.
